main.tf

```diff
@@ -4,158 +4,232 @@ data "aws_eks_cluster" "eks" {
+  source                         = "./modules/cert-manager-le-http-issuer"
+  cert_manager_letsencrypt_email = var.cert_manager_helm_config.cert_manager_letsencrypt_email
+}
+
+## CLUSTER AUTOSCALER
+module "cluster-autoscaler" {
+  count               = var.cluster_autoscaler_enabled ? 1 : 0
+  source              = "./modules/cluster-autoscaler"
+  eks_cluster_version = local.eks_cluster_version
+  helm_config = {
+    version = var.cluster_autoscaler_chart_version
+    values  = var.cluster_autoscaler_helm_config
+  }
+  manage_via_gitops = var.argocd_manage_add_ons
+  addon_context     = local.addon_context
+}
+
+## COREDNS HPA
+module "coredns_hpa" {
+  count       = var.coredns_hpa_enabled ? 1 : 0
+  source      = "./modules/core-dns-hpa" # Replace with the actual path to your module
+  helm_config = var.coredns_hpa_helm_config
+}
+
+## EXTERNAL SECRETS
+module "external-secrets" {
+  count                                 = var.external_secrets_enabled ? 1 : 0
+  source                                = "./modules/external-secret"
+  helm_config                           = var.external_secrets_helm_config
+  manage_via_gitops                     = var.argocd_manage_add_ons
+  addon_context                         = local.addon_context
+  irsa_policies                         = var.external_secrets_irsa_policies
+  external_secrets_ssm_parameter_arns   = var.external_secrets_ssm_parameter_arns
+  external_secrets_secrets_manager_arns = var.external_secrets_secrets_manager_arns
+}
+
+## NGINX INGRESS
+module "ingress-nginx" {
+  count             = var.ingress_nginx_enabled ? 1 : 0
+  depends_on        = [module.aws_vpc_cni, module.service-monitor-crd]
+  source            = "./modules/ingress-nginx"
+  helm_config       = var.ingress_nginx_config
+  manage_via_gitops = var.argocd_manage_add_ons
+  addon_context     = local.addon_context
+  ip_family         = data.aws_eks_cluster.eks.kubernetes_network_config[0].ip_family
+
+  # Template values for ingress-nginx and private-internal-nginx
+  namespace              = var.ingress_nginx_config.namespace
+  enable_private_nlb     = var.enable_private_nlb
+  ingress_class_name     = var.enable_private_nlb ? "internal-${var.ingress_nginx_config.ingress_class_name}" : var.ingress_nginx_config.ingress_class_name
+  enable_service_monitor = var.ingress_nginx_config.enable_service_monitor
+}
+
+## KARPENTER
+module "karpenter" {
+  count                     = var.karpenter_enabled ? 1 : 0
+  source                    = "./modules/karpenter"
+  worker_iam_role_name      = var.worker_iam_role_name
+  eks_cluster_name          = var.eks_cluster_name
+  helm_config               = var.karpenter_helm_config
+  irsa_policies             = var.karpenter_irsa_policies
+  node_iam_instance_profile = var.karpenter_node_iam_instance_profile
+  manage_via_gitops         = var.argocd_manage_add_ons
+  addon_context             = local.addon_context
+}
+
+## Karpenter-provisioner
+module "karpenter-provisioner" {
```

Comment:
Are we still using karpenter provisioning module from local modules here? or we can directly refer our hosted modules.